UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetGenerateFilesForProductIdentifierWithIncorrectValueForIsUnencrytedParameter.cs

```diff
@@ -0,0 +1,124 @@
+            Assert.AreEqual(200, (int)apiScsResponse.StatusCode, $"Incorrect status code is returned {apiScsResponse.StatusCode}, instead of the expected status 200.");
+
+            var apiScsResponseData = await apiScsResponse.ReadAsTypeAsync<ScsProductResponseModel>();
+
+            FileContentHelper.CheckCatalogueFileContent(Path.Combine(DownloadedFolderPath, Config.ExchangeSetEncRootFolder, Config.ExchangeSetCatalogueFile), apiScsResponseData);
+        }
+
+        //PBI:139801 : ESS API : Create and add optional parameter IsUnencrypted, add validation and Update Swagger Doc
+        [Test]
+        [Category("QCOnlyTest-AIODisabled")]
+        public void WhenICallProductIdentifiersApiWithIncorrectValueForIsUnencryptedParameter_ThenASerialEncFileIsGenerated()
+        {
+            var checkFile = FssBatchHelper.CheckforFileExist(DownloadedFolderPath, Config.ExchangeSetSerialEncFile);
+            Assert.IsTrue(checkFile, $"{Config.ExchangeSetSerialEncFile} File not Exist in the specified folder path : {DownloadedFolderPath}");
+
+            //Verify Serial.Enc file content
+            FileContentHelper.CheckSerialEncFileContent(Path.Combine(DownloadedFolderPath, Config.ExchangeSetSerialEncFile));
+        }
+
+        //PBI:139801 : ESS API : Create and add optional parameter IsUnencrypted, add validation and Update Swagger Doc
+        [Test]
+        [Category("QCOnlyTest-AIODisabled")]
+        public async Task WhenICallProductIdentifiersApiWithIncorrectValueForIsUnencryptedParameter_ThenEncFilesAreDownloaded()
+        {
+            //Get the product details form sales catalog service
+            var apiScsResponse = await ScsApiClient.GetProductIdentifiersAsync(Config.ExchangeSetProductType, DataHelper.GetProductIdentifiers(), ScsJwtToken);
+            Assert.AreEqual(200, (int)apiScsResponse.StatusCode, $"Incorrect status code is returned {apiScsResponse.StatusCode}, instead of the expected status 200.");
+
+            var apiScsResponseData = await apiScsResponse.ReadAsTypeAsync<ScsProductResponseModel>();
+
+            foreach (var product in apiScsResponseData.Products)
+            {
+                var productName = product.ProductName;
+                var editionNumber = product.EditionNumber;
+
+                //Enc file download verification
+                foreach (var updateNumber in product.UpdateNumbers)
+                {
+                    await FileContentHelper.GetDownloadedEncFilesAsync(Config.FssConfig.BaseUrl, Path.Combine(DownloadedFolderPath, Config.ExchangeSetEncRootFolder), productName, editionNumber, updateNumber, FssJwtToken);
+
+                }
+            }
+        }
+
+        [OneTimeTearDown]
+        public async Task GlobalTeardown()
+        {
+            //Clean up downloaded files/folders   
+            FileContentHelper.DeleteDirectory(Config.ExchangeSetFileName);
+
+            if (cleanUpBatchIdList != null && cleanUpBatchIdList.Count > 0)
```

Comment:
```suggestion
           if (cleanUpBatchIdList?.Count > 0)
```
please try the above suggestion and update wherever used.

Reply:
Resolved. Thanks